digits/extensions/view/imageSegmentationGL/view.py

```diff
@@ -0,0 +1,190 @@
+            cmap = mpl.colors.ListedColormap(palette)
+            self.map = mpl.pyplot.cm.ScalarMappable(norm=norm, cmap=cmap)
+        elif kwargs['colormap'] == 'paired':
+            cmap = mpl.pyplot.cm.get_cmap('Paired')
+            self.map = mpl.pyplot.cm.ScalarMappable(norm=None, cmap=cmap)
+        elif kwargs['colormap'] == 'none':
+            self.map = None
+        else:
+            raise ValueError("Unknown color map option: %s" % kwargs['colormap'])
+
+        # memorize class labels
+        if 'class_labels' in dataset.extension_userdata:
+            self.class_labels = dataset.extension_userdata['class_labels']
+        else:
+            self.class_labels = None
+
+        # create array to memorize all classes we found in labels
+        self.found_classes = np.array([])
+
+    @staticmethod
+    def get_config_form():
+        return ConfigForm()
+
+    @staticmethod
+    def get_config_template(form):
+        """
+        parameters:
+        - form: form returned by get_config_form(). This may be populated
+        with values if the job was cloned
+        returns:
+        - (template, context) tuple
+          - template is a Jinja template to use for rendering config options
+          - context is a dictionary of context variables to use for rendering
+          the form
+        """
+        extension_dir = os.path.dirname(os.path.abspath(__file__))
+        template = open(
+            os.path.join(extension_dir, CONFIG_TEMPLATE), "r").read()
+        return (template, {'form': form})
+
+    @override
+    def get_header_template(self):
+        """
+        Implements get_header_template() method from view extension interface
+        """
+        extension_dir = os.path.dirname(os.path.abspath(__file__))
+        template = open(
+            os.path.join(extension_dir, HEADER_TEMPLATE), "r").read()
+
+        # show legend
+        legend = []
+        for cls in self.found_classes:
+            # create small square image and fill it with the color
+            # associated with the category ID
+            if self.map:
+                rgb_color = self.map.to_rgba([cls])[0, :3]*255
+                image = np.zeros(shape=(50, 50, 3))
+                image[:, :] = rgb_color
+            else:
+                image = np.full(shape=(50, 50), fill_value=cls)
+            image = image.astype('uint8')
+            image = digits.utils.image.embed_image_html(image)
+            text = "Class #%d" % cls
+            if self.class_labels:
+                text = "%s (%s)" % (text, self.class_labels[int(cls)])
+            legend.append({'image': image, 'text': text})
+        return template, {'legend': legend}
+
+    @override
+    def get_ng_templates(self):
+        """
+        Implements get_ng_templates() method from view extension interface
+        """
+        extension_dir = os.path.dirname(os.path.abspath(__file__))
+        header = open(os.path.join(extension_dir, APP_BEGIN_TEMPLATE), "r").read()
```

Comment:
yes, because if the extension comes from a 3rd party plug-in (outside of DIGITS project) we don't know how to construct the path to the resource files.